WordPress/gutenberg
# [RNMobile] Social icons

## Description

~~**NOTE**: That PR should be merged after https://github.com/WordPress/gutenberg/pull/23299 since it's using the `LinkSettings` component coming from there.~~

~~**NOTE**: That PR should be merged after https://github.com/WordPress/gutenberg/pull/23465~~


That PR introduces a new block called `Social Icon`.

* Ref to gutenberg-mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2361

Testing:

* ios: https://github.com/wordpress-mobile/WordPress-iOS/pull/14259
* android: https://github.com/wordpress-mobile/WordPress-Android/pull/12124

## How has this been tested?

1. Open mobile app
2. Add `Social Icons` block

Expect: Social Icons is added with 4 icons, where WordPress icon is already fulfilled with the link and it's active _(has product color)_

3. Press inactive icon eg. `Facebook` icon
4. Press `link` icon in mobile toolbar above the keyboard
5. Fill `URL` eg. facebook.com and `Label`
6. Close sheet

Expect: Facebook social icon is active _(has product color)_

7. Deselect `Social Icons` 

Expect: Only active social icons left

8. Open post preview
9. Press the icon and check if forwarding to the link

## Screenshots 

* light mode

<details>

active | inactive
--- | ---
<img width="493" alt="Screenshot 2020-06-05 at 16 32 44" src="https://user-images.githubusercontent.com/22746080/83888733-880a0d00-a74a-11ea-89c3-2fc7fdfd062c.png"> | <img width="449" alt="Screenshot 2020-06-05 at 16 33 10" src="https://user-images.githubusercontent.com/22746080/83888736-89d3d080-a74a-11ea-9075-a1418c64b8ef.png">

</details>

* dark mode

<details>

active | inactive
--- | ---
<img width="493" alt="Screenshot 2020-06-05 at 16 32 25" src="https://user-images.githubusercontent.com/22746080/83888890-bd165f80-a74a-11ea-8091-8a09f4be54bb.png"> | <img width="493" alt="Screenshot 2020-06-05 at 16 33 18" src="https://user-images.githubusercontent.com/22746080/83888896-be478c80-a74a-11ea-8b2d-c5fbd1d1f894.png">

</details>

* action sheet

<details>

android | ios
--- | ---
<img width="427" alt="Screenshot 2020-06-08 at 15 50 22" src="https://user-images.githubusercontent.com/22746080/84038139-d401bf80-a99f-11ea-908a-305428331225.png"> | <img width="493" alt="Screenshot 2020-06-08 at 15 54 12" src="https://user-images.githubusercontent.com/22746080/84038507-54282500-a9a0-11ea-81c9-8444758e3899.png">

</details>

* URL sheet

<details>

android | ios
--- | ---
<img width="427" alt="Screenshot 2020-06-09 at 10 29 57" src="https://user-images.githubusercontent.com/22746080/84125051-4b852c80-aa3c-11ea-9d9f-783b5ea246b7.png"> | <img width="493" alt="Screenshot 2020-06-09 at 10 27 34" src="https://user-images.githubusercontent.com/22746080/84125038-47590f00-aa3c-11ea-9e1d-14c5f30dce09.png">

</details>

## Types of changes

New feature: `Social Icons` block

## Checklist:
